src/main/java/com/google/jspecify/nullness/NullSpecAnnotatedTypeFactory.java

```diff
@@ -664,7 +664,11 @@ && isNullnessSubtype(subtype, ((AnnotatedTypeVariable) supertype).getLowerBound(
         return true;
       }
       if (subtype.getKind() == TYPEVAR && supertype.getKind() == TYPEVAR) {
+        // Work around wonkyness of CF override checks.
         AnnotatedTypeVariable subTV = (AnnotatedTypeVariable) subtype;
+        if (isCapturedTypeVariable(subTV.getUnderlyingType())) {
+          subTV = (AnnotatedTypeVariable) subTV.getUpperBound();
```

Comment:
It looks like we'd need to check that assumption:

> Exception: java.lang.ClassCastException: class org.checkerframework.framework.type.AnnotatedTypeMirror$AnnotatedDeclaredType cannot be cast to class org.checkerframework.framework.type.AnnotatedTypeMirror$AnnotatedTypeVariable (org.checkerframework.framework.type.AnnotatedTypeMirror$AnnotatedDeclaredType and org.checkerframework.framework.type.AnnotatedTypeMirror$AnnotatedTypeVariable are in unnamed module of loader java.net.URLClassLoader @1ac83ad9); java.lang.ClassCastException: class org.checkerframework.framework.type.AnnotatedTypeMirror$AnnotatedDeclaredType cannot be cast to class org.checkerframework.framework.type.AnnotatedTypeMirror$AnnotatedTypeVariable (org.checkerframework.framework.type.AnnotatedTypeMirror$AnnotatedDeclaredType and org.checkerframework.framework.type.AnnotatedTypeMirror$AnnotatedTypeVariable are in unnamed module of loader java.net.URLClassLoader @1ac83ad9)

That's probably at https://github.com/google/auto/blob/68d99dc75b8379ca14e56c8c4f10493b1b0cb38d/common/src/main/java/com/google/auto/common/SuperficialValidation.java#L237.

I think we're fine just not worrying about trying to handle the captured-type-variable case at all for now. But if you want to look into it more, I can get you more information tomorrow. Let me know.

Reply:
Ah! I added a check to ensure the cast is safe.

Reply:
I've confirmed (a) that it eliminates all the crashes I was seeing and (b) that the overall change does address the `? extends E` errors I was seeing.